tests/sles4sap/monitoring_services.pm

```diff
@@ -0,0 +1,156 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Installation and basic tests of exporters for SLES4SAP monitoring
+# Maintainer: Loic Devulder <ldevulder@suse.com>
+
+use base "sles4sap";
+use testapi;
+use strict;
+use warnings;
+use lockapi;
+use Utils::Systemd qw(systemctl);
+use utils qw(zypper_call);
+use hacluster qw(get_cluster_name is_node wait_until_resources_started);
+
+sub run {
+    my ($self)        = @_;
+    my $cluster_name  = get_cluster_name;
+    my $instance_sid  = get_required_var('INSTANCE_SID');
+    my $instance_type = get_required_var('INSTANCE_TYPE');
+    my $instance_id   = get_required_var('INSTANCE_ID');
+    my $rsc_id        = "${instance_sid}_${instance_type}${instance_id}";
+
+    $self->select_serial_terminal;
+
+    # Configure HA Exporter
+    if (get_var('HA_CLUSTER')) {
+        my $ha_exporter_config = 'ha_cluster_exporter';
+
+        # Install needed packages
+        zypper_call "in prometheus-$ha_exporter_config";
+
+        # Get the IP port and start exporter
+        my ($ha_exporter_port) = script_output("awk '/port:/ { print \$NF }' /etc/$ha_exporter_config /usr/etc/$ha_exporter_config 2>/dev/null", proceed_on_failure => 1) =~ /(\d+)/;
+        $ha_exporter_port //= 9664;
```

Comment:
Not sure if it would be better here to use `||=` instead of `//=`. For example, a port value of 0 in the config file would be evaluated as true by `//=` and not change the value on `$ha_export_port`, and a port value of 0 further down the test would not yield valid results, but in a configuration file 0 could mean "use the default".

Reply:
Thanks, in that case yes `||=` could be better.